packages/core/src/types.ts

```diff
@@ -1014,6 +1014,20 @@ export interface InterpreterOptions {
 
 export type AnyInterpreter = Interpreter<any, any, any>;
 
+/**
+ * Represents the `Interpreter` type of a given `MachineNode`.
+ *
+ * @typeParam TM - the machine to infer the interpreter's types from
+ */
+export type InterpreterOf<TM extends MachineNode> = TM extends MachineNode<
```

Comment:
Heads up @Andarist - added this type while refactoring to make the tests pass, and it's actually really useful!